wireapp/kalium
# fix: Duplicated calling system messages (AR-2744)

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Issue 1 : When you start a 1:1 call and end it without accepting on the other side, “You called” system message is displayed twice

Issue 2: When you reject an incoming call and the caller terminates the call without being accepted, “Someone called” is displayed twice

### Causes (Optional)

We were updating the call two times when the call is ended or rejected: The first time was in `OnCloseCall` callback and the second was in `endCall()`/`rejectCall()` method
Also we were not checking the `reason` field in `OnCloseCall` callback

### Solutions

Remove duplicated code and Implement missing parts
Also some clean up is performed: Removed `OnMissedCall` callback since it never occurs 

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:


### Testing

#### Test Coverage (Optional)


#### How to Test

Issue 1:
- Start a 1:1 call and end it without accepting on the other side.
Issue 2:
- Reject an incoming 1:1 call and wait until the caller terminates it (The call should not be accepted anywhere)

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
